src/main/java/org/sonarsource/auth/github/UserIdentityFactoryImplBeforeSQ72.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * GitHub Authentication for SonarQube
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.auth.github;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.sonar.api.server.authentication.UserIdentity;
+
+import static org.sonarsource.auth.github.UserIdentityGenerator.generateLogin;
+import static org.sonarsource.auth.github.UserIdentityGenerator.generateName;
+
+/**
+ * The goal of this class is to be able to generate an identity on SonarQube version lesser than 7.2
+ */
+public class UserIdentityFactoryImplBeforeSQ72 implements UserIdentityFactory {
+
+  private final GitHubSettings settings;
+
+  public UserIdentityFactoryImplBeforeSQ72(GitHubSettings settings) {
+    this.settings = settings;
+  }
+
+  @Override
+  public UserIdentity create(GsonUser user, @Nullable String email, @Nullable List<GsonTeams.GsonTeam> teams) {
```

Comment:
No need to duplicate the class, you can keep only `UserIdentityFactoryImpl` by injecting `org.sonar.api.SonarRuntime`:

```
UserIdentity.Builder builder = UserIdentity.builder();
// provider id is not set as this method was added in SonarQube 7.2
if (runtime.getApiVersion().isGreaterThanOrEqual(Version.create(7, 2))) {
   builder.setProviderId(user.getId());
 }
builder.setProviderLogin(user.getLogin())
```